src/providers/storage/s3.js

```diff
@@ -5,9 +5,23 @@ const s3 = (formio) => ({
     return XHR.upload(formio, 's3', (xhr, response) => {
       response.data.fileName = fileName;
       response.data.key = XHR.path([response.data.key, dir, fileName]);
+      const setXhrHeaders = (xhrToSet) => {
```

Comment:
I actually see this being valuable for all XHR file upload requests and not just S3.  Can we move this into the ```XHR.upload``` method?

Reply:
Absolutely!